insonmnia/dealer/match.go

```diff
@@ -0,0 +1,53 @@
+package dealer
+
+import (
+	"github.com/sonm-io/core/proto"
+)
+
+type Matcher interface {
+	// Match takes orders found on market, returns most profitable order to deal with
+	Match(orders []*sonm.Order) (*sonm.Order, error)
+}
+
+// bidMatcher matches the cheapest order to deal
+type bidMatcher struct{}
+
+func NewBidMatcher() Matcher {
+	return &bidMatcher{}
+}
+
+func (m *bidMatcher) Match(orders []*sonm.Order) (*sonm.Order, error) {
+	if len(orders) == 0 {
+		return nil, nil
+	}
+
+	var min = orders[0]
+	for _, o := range orders {
+		if o.PricePerSecond.Cmp(min.PricePerSecond) < 0 {
+			min = o
+		}
+	}
+
+	return min, nil
+}
+
+type askMatcher struct{}
+
+func NewAskMatcher() Matcher {
+	return &askMatcher{}
+}
+
+func (m *askMatcher) Match(orders []*sonm.Order) (*sonm.Order, error) {
+	if len(orders) == 0 {
+		return nil, nil
```

Comment:
I think we should return error in such case.

Reply:
Ouch, fixed this into #513